crypto/modes/modes_lcl.h

```diff
@@ -88,9 +94,13 @@ _asm mov eax, val _asm bswap eax}
 #if defined(BSWAP4) && !defined(STRICT_ALIGNMENT)
 # define GETU32(p)       BSWAP4(*(const u32 *)(p))
 # define PUTU32(p,v)     *(u32 *)(p) = BSWAP4(v)
+# define GETUU32(p)      BSWAP4(*(const ua_u32 *)(p))
+# define PUTUU32(p,v)    *(ua_u32 *)(p) = BSWAP4(v)
```

Comment:
Does one actually have to have two? I mean why can't already [GET|PUT]U32 use alternative cast? What am I missing?

Reply:
for intel they are of course identical.
for arm the UU version generates 4 byte accesses.
So since most of the GETU32/PUTU32 are aligned they do not need that.

Reply:
But armv4 shouldn't have STRICT_ALIGNED... Ah! You ought to refer to that question about ctr128.c, right? I haven't looked into it yet...

Reply:
Aehm, I overlooked the fact that STRICT_ALIGNED uses a variant of GETU32 with single byte access.
So, I think the GETU32 can use the unaligned type, without making arm code worse than it is already.